Lib/test/test_sqlite3/test_dump.py

```diff
@@ -49,6 +51,59 @@ def test_table_dump(self):
         [self.assertEqual(expected_sqls[i], actual_sqls[i])
             for i in range(len(expected_sqls))]
 
+    def test_dump_autoincrement(self):
+        expected_sqls = [
+            "CREATE TABLE \"posts\" (id int primary key);",
+            "INSERT INTO \"posts\" VALUES(0);",
+            "CREATE TABLE \"tags\" (id integer primary key autoincrement, tag varchar(256), post int references posts);",
+            "INSERT INTO \"tags\" VALUES(NULL,'test',0);",
+            "CREATE TABLE \"tags2\" (id integer primary key autoincrement, tag varchar(256), post int references posts);"
+        ]
+
+        for sql_statement in expected_sqls:
+            self.cu.execute(sql_statement)
+        iterdump = self.cx.iterdump()
+        actual_sqls = [sql_statement for sql_statement in iterdump]
+
+        # the NULL value should now be automatically be set to 1
+        expected_sqls[3] = expected_sqls[3].replace("NULL", "1")
+        expected_sqls.insert(0, "BEGIN TRANSACTION;")
+        expected_sqls.extend([
+            "DELETE FROM \"sqlite_sequence\";",
+            "INSERT INTO \"sqlite_sequence\" VALUES('tags',1);",
+            "COMMIT;"
+        ])
+
+        self.assertEqual(expected_sqls, actual_sqls)
+
+    def test_dump_autoincrement_create_new_db(self):
+        old_db = [
+            "BEGIN TRANSACTION ;",
+            "CREATE TABLE \"posts\" (id int primary key);",
+            "INSERT INTO \"posts\" VALUES(0);",
+            "CREATE TABLE \"tags\" (id integer primary key autoincrement, tag varchar(256), post int references posts);",
+            "CREATE TABLE \"tags2\" (id integer primary key autoincrement, tag varchar(256), post int references posts);"
+        ]
+        for i in range(1, 10):
+            old_db.append("INSERT INTO \"tags\" VALUES(NULL,'test{0}',0);".format(i))
+        for i in range(1, 5):
+            old_db.append("INSERT INTO \"tags2\" VALUES(NULL,'test{0}',0);".format(i))
+        old_db.append("COMMIT;")
+
+        self.cu.executescript("".join(old_db))
+        query = "".join(self.cx.iterdump())
+
+        with memory_database() as cx2:
+            cx2.executescript(query)
+            cu2 = cx2.cursor()
+
+            self.assertEqual(cu2.execute("SELECT \"seq\""
+                                         "FROM \"sqlite_sequence\""
+                                         "WHERE \"name\" == \"tags\"").fetchall()[0][0], 9)
```

Comment:
This will not work if SQLite is compiled with, for example, `-DSQLITE_DQS=0`; you need to use single quotes around string literals. Ditto for the other SELECT.

Reply:
Thanks for the hint, I was not aware of that. It should be fixed in the newest version :+1: